src/main/java/wooteco/subway/domain/path/Path.java

```diff
@@ -1,64 +1,61 @@
 package wooteco.subway.domain.path;
 
+import java.util.ArrayList;
 import java.util.List;
-import java.util.Objects;
-import wooteco.subway.domain.path.fare.Fare;
+import java.util.stream.Collectors;
+import wooteco.subway.domain.fare.Fare;
+import wooteco.subway.domain.line.LineExtraFare;
+import wooteco.subway.domain.section.Section;
 import wooteco.subway.domain.station.Station;
 
 public class Path {
 
-    private final List<Station> stations;
-    private final int distance;
-    private final int fare;
+    private static final String SELF_LOOP_EXCEPTION = "출발점과 도착점은 동일할 수 없습니다.";
 
-    public Path(List<Station> stations, int distance, int fare) {
-        this.stations = stations;
-        this.distance = distance;
-        this.fare = fare;
+    private final Station source;
+    private final List<Section> route;
+
+    public Path(Station source, Station target, Navigator navigator) {
+        validateNonSelfLoop(source, target);
+        this.source = source;
+        this.route = navigator.calculateShortestPath(source, target);
     }
 
-    public static Path of(List<Station> stations, int distance) {
-        Fare fare = Fare.of(distance);
-        return new Path(stations, distance, fare.getValue());
+    private static void validateNonSelfLoop(Station source, Station target) {
+        if (source.equals(target)) {
+            throw new IllegalArgumentException(SELF_LOOP_EXCEPTION);
+        }
     }
 
-    public List<Station> getStations() {
+    public List<Station> toStations() {
+        List<Station> stations = new ArrayList<>();
+        stations.add(source);
+        for (Section section : route) {
+            Station lastStation = stations.get(stations.size() - 1);
+            stations.add(section.getOppositeEnd(lastStation));
+        }
         return stations;
     }
 
     public int getDistance() {
-        return distance;
-    }
-
-    public int getFare() {
-        return fare;
+        return route.stream()
+                .mapToInt(Section::getDistance)
+                .sum();
     }
 
-    @Override
-    public boolean equals(Object o) {
-        if (this == o) {
-            return true;
-        }
-        if (o == null || getClass() != o.getClass()) {
-            return false;
-        }
-        Path path = (Path) o;
-        return distance == path.distance
-                && fare == path.fare
-                && Objects.equals(stations, path.stations);
+    public List<Long> getPassingLineIds() {
+        return route.stream()
+                .map(Section::getLineId)
+                .distinct()
+                .collect(Collectors.toList());
     }
 
-    @Override
-    public int hashCode() {
-        return Objects.hash(stations, distance, fare);
-    }
+    public int calculateFare(List<LineExtraFare> extraFares, int age) {
```

Comment:
fare 변수를 새로 교체할 수도 있지만 요런식으로 써볼 수도 있겠네요

```suggestion
    public int calculateFare(List<LineExtraFare> extraFares, int age) {
        return new Fare()
                .applyDistanceOverFarePolicies(getDistance())
                .applyMaximumLineExtraFare(extraFares)
                .applyAgeDiscountPolicy(age)
                .toInt();
    }
```

Reply:
그러네요. 데코레이터 패턴으로 추구했던 것들이 전부 메서드 체이닝으로 되는거였네요.